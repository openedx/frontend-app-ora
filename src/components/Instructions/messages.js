import { defineMessages } from '@edx/frontend-platform/i18n';

import { stepNames } from 'constants/index';

const messages = defineMessages({
  [stepNames.submission]: {
    defaultMessage: 'Enter your response to the prompt. Your work will save automatically and you can return to complete your response at any time before the due date. After you submit your response, you cannot edit it.',
    description: 'Submission step instructions',
    id: 'frontend-app-ora.instructions.submission',
  },
  [stepNames.studentTraining]: {
    defaultMessage: 'Before you begin to assess your peers\' responses, you\'ll learn how to complete peer assessments by reviewing responses that instructors have already assessed. If you select the same options for the response that the instructor selected, you\'ll move to the next step. If you don\'t select the same options, you\'ll review the response and try again.',
    description: 'StudentTraining step instructions',
    id: 'frontend-app-ora.instructions.studentTraining',
  },
  [stepNames.self]: {
    defaultMessage: 'Assess your response',
    description: 'Self assessment step instructions',
    id: 'frontend-app-ora.instructions.selfAssessment',
  },
  [stepNames.peer]: {
    defaultMessage: 'Read and assess the following response from one of your peers.',
    description: 'Peer assessment step instructions',
    id: 'frontend-app-ora.instructions.peerAssessment',
  },
  [stepNames.done]: {
    defaultMessage: 'You have successfully completed this problem and received a {earned}/{possible}.',
    description: 'Graded step instructions',
    id: 'frontend-app-ora.instructions.done',
  },
  instructions: {
    defaultMessage: 'Instructions',
    description: 'Instructions header',
    id: 'frontend-app-ora.instructions.header',
  },
});

export default messages;
