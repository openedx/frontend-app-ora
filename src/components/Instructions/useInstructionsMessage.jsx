import { useIntl } from '@edx/frontend-platform/i18n';

import { stepNames, stepStates } from 'constants/index';

import { useGlobalState } from 'hooks/app';
import { useViewStep } from 'hooks/routing';

import messages from './messages';

const useInstructionsMessage = () => {
  const { formatMessage } = useIntl();
  const viewStep = useViewStep();
  const {
    activeStepName,
    effectiveGrade,
    stepState,
  } = useGlobalState();
  const stepName = (viewStep === stepNames.xblock) ? activeStepName : viewStep;
  if (stepState !== stepStates.inProgress || stepName === stepNames.staff) {
    return null;
  }
  if (stepName === stepNames.done) {
    return formatMessage(messages[stepNames.done], effectiveGrade);
  }
  return messages[stepName] && formatMessage(messages[stepName]);
};

export default useInstructionsMessage;
