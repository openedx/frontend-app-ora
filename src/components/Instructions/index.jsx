import React from 'react';

import { useIntl } from '@edx/frontend-platform/i18n';

import useInstructionsMessage from './useInstructionsMessage';

import messages from './messages';

const Instructions = () => {
  const { formatMessage } = useIntl();
  const message = useInstructionsMessage();
  if (!message) {
    return null;
  }
  return (
    <div className="py-4">
      <p className="mb-0">
        <strong>{formatMessage(messages.instructions)}: </strong>
        {message}
      </p>
    </div>
  );
};
export default Instructions;
