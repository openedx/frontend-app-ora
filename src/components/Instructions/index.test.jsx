import { shallow } from '@edx/react-unit-test-utils';

import useInstructionsMessage from './useInstructionsMessage';

import Instructions from './index';

jest.mock('./useInstructionsMessage', () => jest.fn());

describe('<Instructions />', () => {
  it('render empty on no message', () => {
    useInstructionsMessage.mockReturnValue(null);
    const wrapper = shallow(<Instructions />);
    expect(wrapper.snapshot).toMatchSnapshot();

    expect(wrapper.isEmptyRender()).toBe(true);
  });

  it('render message', () => {
    useInstructionsMessage.mockReturnValue('arbitrarilyInstructionsMessage');
    const wrapper = shallow(<Instructions />);
    expect(wrapper.snapshot).toMatchSnapshot();
  });
});
