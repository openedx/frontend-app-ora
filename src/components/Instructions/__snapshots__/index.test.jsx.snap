// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Instructions /> render empty on no message 1`] = `null`;

exports[`<Instructions /> render message 1`] = `
<div
  className="py-4"
>
  <p
    className="mb-0"
  >
    <strong>
      Instructions
      : 
    </strong>
    arbitrarilyInstructionsMessage
  </p>
</div>
`;
